cmd/rancher-desktop-guestagent/main.go

```diff
@@ -75,29 +80,60 @@ func main() {
 
 	group, ctx := errgroup.WithContext(context.Background())
 
-	if *enableDocker {
+	if *enablePriviledService && *enableIptables {
+		log.Fatal("error enabling Privileged Service mode and iptables, you can only enable one.")
+	}
+
+	if *enablePriviledService {
+		if !*enableContainerd && !*enableDocker {
+			log.Fatal("enabling Privileged Service mode requires either Docker or Containerd enabled.")
+		}
+
+		if *enableContainerd && *enableDocker {
+			log.Fatal("enabling Privileged Service mode requires enabling either Docker or Containerd not both.")
```

Comment:
See above. What if I _didn't_ have `-privilegedService`; can I have `-docker -containerd` in that case? If not, the check should be moved out of the `if *privilegedService` check.

Reply:
Both docker and container are subcommands to Privileged service, so non should be activated w/o Privileged service flag.